packages/subgraph/src/entities/SystemState.ts

```diff
@@ -135,15 +145,22 @@ export function updateSystemStateByTroveChange(troveChange: TroveChange): void {
   let operation = troveChange.troveOperation;
 
   if (isBorrowerOperation(operation) || isRedemption(operation)) {
-    systemState.totalCollateral += troveChange.collateralChange;
-    systemState.totalDebt += troveChange.debtChange;
+    systemState.totalCollateral = systemState.totalCollateral.plus(troveChange.collateralChange);
+    systemState.totalDebt = systemState.totalDebt.plus(troveChange.debtChange);
   } else if (isLiquidation(operation)) {
-    // TODO gas compensation
+    let collateral = troveChange.collateralBefore;
+    let debt = troveChange.debtBefore;
+    let collateralGasCompensation = collateral
+      .div(DECIMAL_COLLATERAL_GAS_COMPENSATION_DIVISOR)
+      .truncate(DECIMAL_PRECISION);
+
+    systemState.totalCollateral = systemState.totalCollateral.minus(collateralGasCompensation);
+
     if (!isRecoveryModeLiquidation(operation) || troveChange.collateralRatioBefore > DECIMAL_ONE) {
       tryToOffsetWithTokensFromStabilityPool(
         systemState,
-        -troveChange.collateralChange,
-        -troveChange.debtChange
+        collateral.minus(collateralGasCompensation),
+        debt
```

Comment:
I don’t get how this is equivalent. Letting aside the gas compensation, it seems that now we are using “before” values, instead of “change” values. What am I missing?

Reply:
I originally used the change values, because back then there used to be partial liquidation, which didn't completely erase the Trove. With those out of the picture, collateralChange and debtChange are always equal to -collateral and -debt, so it's simpler to just use collateral and debt.

Reply:
Makes sense, thanks!